.github/workflows/build_and_test.yml

```
@@ -11256,3 +11289,40 @@ jobs:
     - uses: ammaraskar/gcc-problem-matcher@0.1
     - name: Build
       run: cd OpenDDS && docker build -f .github/workflows/build_u14_gcc4.Dockerfile .
+
+  openssl3_u20:
+
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: checkout openssl 3.0.2
+      run: |
+        git clone --branch openssl-3.0.2 --depth 1 https://github.com/openssl/openssl.git
+        cd openssl
+        echo "OPENSSL_COMMIT=$(git rev-parse HEAD)" >> $GITHUB_ENV
+    - name: Cache Artifact
+      id: cache-artifact
+      uses: actions/cache@v2
+      with:
+        path: ${{ github.job }}.tar.xz
+        key: c02_${{ github.job }}_${{ env.OPENSSL_COMMIT }}
+    - name: openssl 3.0.2
```

To make the script happy:

```suggestion
    - name: Build openssl 3.0.2
```

---

@iguessthislldo What does the current lint failure mean?

---

Ah, I didn't think about it wanting a problem matching. The script a bit wonky here because of that and without `-j` it's matching `make install`, which doesn't count as a build command as far it's concerned. Now that I think about it, the script probably should just get an exception for OpenSSL like it [already has for Google Test and ACE/TAO](https://github.com/objectcomputing/OpenDDS/blob/9c284b4c8c8ae5a10d14f5824306cad12dcb4040/.github/workflows/lint_build_and_test.pl#L61-L67). Locally I removed the problem matcher and added:

```
      elsif ($step->{name} =~ /build openssl/i) {
        $avoids_problem_matcher = 1;
      }
```

to the script and the script ran successfully.